test/rules/always_put_control_body_on_new_line.dart

```diff
@@ -19,6 +19,13 @@ testIfElse() {
     return; // OK
   else return; // LINT
 
+  if (false) {
+  }
+  else if (false) { // OK
```

Comment:
I checked with @sethladd, and as far as I can tell there are no plans to have a second formatter. Some of the code in the Flutter SDK won't be formatted using the existing formatter because of those stylistic differences, but Flutter users are encouraged to use `dartfmt`. That means that this shouldn't be viewed as a short-term "helper", but as a long-term aid to maintaining a different formatting style.

In light of that, I think it's important that any lints we write tolerate (as much as possible) the formatter's style as well. I don't think there's actually any conflict in this case. The formatter isn't trying to put the control body on the same line, and the formatting changes I suggested shouldn't cause any lints of this kind to be generated. (And to be clear, I'm suggesting that we include both styles in the test so that we know the lint is robust across non-critical differences.)